tools/automation/sections/device_section.py

```diff
@@ -0,0 +1,212 @@
+"""
+  A collection of classes to represent the device section of the YAML.
+
+  They are responsible for filling up the section with data.
+
+  They also hold the logic for how to read the input files in order to
+  retrieve the correct data for a device.
+"""
+
+import re
+import abc
+
+from building_blocks.device import Device
+
+"""
+  Base class for the device section.
+"""
+class DeviceSection:
+
+  def __init__(self, files):
+    self._devices = []
+
+  # Logic for reading the excel data to fill up device data.
+  @abc.abstractmethod
+  def _create_device(self, data):
+    pass
+
+  # Logic for filling up device list goes here.
+  @abc.abstractmethod
+  def _populate_devices(self):
+    pass
+
+  # Call to retrieve the data in the form of a dictionary.
+  def to_dictionary(self):
+    device_dictionary = {}
+
+    for device in self._devices:
+      device_dictionary.update(device.to_dictionary())
+
+    return device_dictionary
+
+
+class DBODeviceSection(DeviceSection):
+  
+  def __init__(self, files):
+    super().__init__(files)
+    self._site_model_sheets = files.site_model_sheets
+    self._site_model_columns = files.SiteModelColumns
+    self._populate_devices()
+
+  def _get_devices_from_string(self, devices):
+    linked_devices = []
+
+    for device in devices.split(","):
+      linked_devices.append(device.strip())
+      
+    return linked_devices
+
+  def _get_units_dict(self, units_string):
+    unit_strings = units_string.split(":")
+    return {unit_strings[0].strip(): unit_strings[1].strip()}
+
+  def _get_states_dict(self, state_string):
+    state_strings = state_string.split()
+    state_dict = {}
+
+    for state in state_strings:
+      state_dict.update({
+        state.split("=")[1].upper(): "\"" + state.split("=")[0].upper() + "\""
+      })
+
+    return state_dict
+
+  def _fill_device_connections(self, row, device):
+    feeds = row[self._site_model_columns.CONNECTION_FEEDS]
+    controls = row[self._site_model_columns.CONNECTION_CONTROLS]
+
+    if not feeds.isspace() and len(feeds) > 0:
+      feeds = self._get_devices_from_string(feeds)
+      for i in feeds:

```

Comment:
nit: "i" seems to imply index here but I think it's supposed to be a connection name, maybe prefer "name" instead